test/e2e/agentgateway_e2e_test.go

```diff
@@ -116,12 +102,88 @@ var _ = Describe("Agent Gateway", Ordered, func() {
 
 	It("should retrieve agent card with correct url", func() {
 		By("retrieving agent card from the gateway")
-		body, err := utils.GetRequest(gatewayUrl + "/mocked-agent-exposed-1/.well-known/agent-card.json")
+		body, statusCode, err := utils.MakeGatewayGet("default", "agent-gateway", 10000,
+			"/mocked-agent-exposed-1/.well-known/agent-card.json")
 		Expect(err).NotTo(HaveOccurred(), "Failed to send GET request to agent-card endpoint")
+		Expect(statusCode).To(Equal(200), "Expected HTTP 200 status code")
 
 		var responseMap map[string]interface{}
 		err = json.Unmarshal(body, &responseMap)
 		Expect(err).NotTo(HaveOccurred(), "Failed to unmarshal agent card response")
-		Expect(responseMap["url"]).To(Equal(gatewayUrl + "/mocked-agent-exposed-1"))
+		Expect(responseMap["url"]).NotTo(BeEmpty(), "Agent card URL should not be empty")
+		Expect(responseMap["url"]).To(ContainSubstring("/mocked-agent-exposed-1"),
+			"Agent card URL should contain the agent path")
+	})
+
+	It("should reload when agent is deleted", func() {
+		By("verifying agent is accessible via agent card")
+		_, statusCode, err := utils.MakeGatewayGet("default", "agent-gateway", 10000,
+			"/mocked-agent-exposed-1/.well-known/agent-card.json")
+		Expect(err).NotTo(HaveOccurred(), "Agent card should be accessible before deletion")
+		Expect(statusCode).To(Equal(200), "Expected HTTP 200 status code of agent")
+
+		By("deleting the agent")
+		err = utils.DeleteAgent("mocked-agent-exposed-1", "default")
+		Expect(err).NotTo(HaveOccurred(), "Failed to delete agent")
+
+		By("verifying deleted agent returns 404")
+		Eventually(func() error {
+			// Make request to check if gateway returns 404 for deleted agent
+			_, statusCode, err := utils.MakeGatewayGet("default", "agent-gateway", 10000,
+				"/mocked-agent-exposed-1/.well-known/agent-card.json")
+			if err != nil {
+				// Connection/network error - retry
+				_, _ = fmt.Fprintf(GinkgoWriter, "Connection error (will retry): %v\n", err)
+				return fmt.Errorf("failed to connect to gateway: %w", err)
+			}
+
+			// Log the received status code
+			_, _ = fmt.Fprintf(GinkgoWriter, "Received HTTP status code: %d\n", statusCode)
```

Comment:
What is all this logging for? Why not simply assert the status code? This way, we know the expected and actual status code with much less code ;)